pkg/policies/policy_groups.go

```diff
@@ -282,3 +307,25 @@ func (pgv *PolicyGroupVerifier) shouldApplyPolicy(ctx context.Context, policyAtt
 
 	return false, nil
 }
+
+// getPolicyName extracts the metadata.name from a PolicyAttachment
+// It handles both embedded and referenced policies by loading the policy spec when needed
+func (pgv *PolicyGroupVerifier) getPolicyName(ctx context.Context, attachment *v1.PolicyAttachment) (string, error) {
+	// Case 1: Embedded policy - direct access
+	if embedded := attachment.GetEmbedded(); embedded != nil {
```

Comment:
`loadPolicySpec` already contains the logic for embedded policies. This first case block wouldn't be needed.